### PR TITLE
Consume terra dev site 1.0

### PR DESCRIPTION
**Pull request by dkasper-was-taken**
### Summary
Migrate code examples and remove terra-site, update tests to new routes and file structure. Additionally, removed terra-i18n-plugin package.

### Deployment URL
https://terra-core-deployed-pr-1542.herokuapp.com/

Thanks for contributing to Terra. 
@cerner/terra

Go ahead... merge it...

### CODE REVIEW

**Comment by mjhenkes** [comment]
1,855 changes

File: packages/terra-table/tests/nightwatch/single-select-table/single-select-table-spec.js [hunk 1]
```diff
@@ -1,119 +0,0 @@
-/* eslint-disable no-unused-expressions */
```
Comment (emilyrohrbough) [inline packages/terra-table/tests/nightwatch/single-select-table/single-select-table-spec.js]: Why did we drop these?

File: packages/terra-table/tests/nightwatch/selectable-table/selectable-table-rows-spec.js [hunk 2]
```diff
@@ -1,74 +0,0 @@
-/* eslint-disable no-unused-expressions */
```
Comment (emilyrohrbough) [inline packages/terra-table/tests/nightwatch/selectable-table/selectable-table-rows-spec.js]: Why did we drop these?

File: packages/terra-table/tests/nightwatch/single-select-table/single-select-table-spec.js (hunk 1 above)
Comment (mjhenkes) [reply packages/terra-table/tests/nightwatch/single-select-table/single-select-table-spec.js]: because we want don't want unused expressions?

**Comment by emilyrohrbough** [comment]
Since you removed the tera-i18n-pluign package, the link on the readme needs removed

File: packages/terra-table/tests/nightwatch/single-select-table/single-select-table-spec.js (hunk 1 above)
Comment (mjhenkes) [reply packages/terra-table/tests/nightwatch/single-select-table/single-select-table-spec.js]: not dropped, just moved to packages/terra-table/nightwatch/single-select-table-spec.js

File: packages/terra-table/tests/nightwatch/selectable-table/selectable-table-rows-spec.js (hunk 2 above)
Comment (mjhenkes) [reply packages/terra-table/tests/nightwatch/selectable-table/selectable-table-rows-spec.js]: not dropped just moved to  packages/terra-table/nightwatch/selectable-table-rows-spec.js

File: dev-site-config/site.config.js [hunk 3]
```diff
@@ -1,33 +1,20 @@
...
-    },
-
-    extensions: {
-      gitHubUrl: 'https://github.com/cerner/terra-core',
     },
   },
 };
 
-export default siteConfig;
+module.exports = siteConfig;
```
Comment (JakeLaCombe) [inline dev-site-config/site.config.js]: Why this change?

File: packages/terra-button/package.json [hunk 4]
```diff
@@ -34,11 +34,13 @@
...
-    "terra-base": "^3.7.0"
+    "terra-base": "^3.7.0",
+    "terra-doc-template": "^1.3.0"
   },
   "dependencies": {
     "classnames": "^2.2.5",
     "prop-types": "^15.5.8",
-    "terra-base": "^3.10.0"
+    "terra-base": "^3.10.0",
+    "terra-icon": "^2.9.0"
```
Comment (bjankord) [inline packages/terra-button/package.json]: Should `terra-doc-template` be added as dependency here too?

Comment (mjhenkes) [reply packages/terra-button/package.json]: It's currently listed as a peer. we'll switch it over to a pure dependency.

File: packages/terra-abstract-modal/src/terra-dev-site/doc/example/AbstractModalCloseOnOutsideClick.jsx [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        <AbstractModal
+          ariaLabel="Modal disable close on outside click"
+          isOpen={this.state.isOpen}
+          closeOnOutsideClick={false}
+          onRequestClose={this.handleCloseModal}
+          classNameModal={styles['fixed-size']}
+        >
+          <div style={{ height: '100%', width: '100%', overflow: 'hidden', backgroundColor: 'lightgreen' }}>
+            <h1>Modal disable close on outside click</h1>
+            <br />
```
Comment (JakeLaCombe) [inline packages/terra-abstract-modal/src/terra-dev-site/doc/example/AbstractModalCloseOnOutsideClick.jsx]: Are these br's necessary for this example? It looks like they are used just for spacing, in which it would be better just to do styles on the paragraphs.

File: packages/terra-abstract-modal/src/terra-dev-site/doc/example/AbstractModalIsOpened.jsx [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+            <br />
+            <p>You can close the modal by:</p>
+            <ul>
+              <li>- Pressing the ESC key</li>
+              <li>- Clicking on the overlay</li>
+              <li>- Clicking on the close button</li>
+            </ul>
+            <br />
+            <p>On smaller screens, the modal will take up the full screen.</p>
+            <p />
```
Comment (JakeLaCombe) [inline packages/terra-abstract-modal/src/terra-dev-site/doc/example/AbstractModalIsOpened.jsx]: Empty paragraph?

File: packages/terra-abstract-modal/src/terra-dev-site/doc/example/AbstractModalCloseOnOutsideClick.jsx (hunk 5 above)
Comment (mjhenkes) [reply packages/terra-abstract-modal/src/terra-dev-site/doc/example/AbstractModalCloseOnOutsideClick.jsx]: These were present in the previous examples, we just moved their location in this PR.

File: packages/terra-abstract-modal/src/terra-dev-site/doc/example/AbstractModalIsOpened.jsx (hunk 6 above)
Comment (mjhenkes) [reply packages/terra-abstract-modal/src/terra-dev-site/doc/example/AbstractModalIsOpened.jsx]: These were present in the previous examples, we just moved their location in this PR.

File: packages/terra-abstract-modal/src/terra-dev-site/test/abstract-modal/AbstractModalTestStyles.scss [hunk 7]
```diff
@@ -0,0 +1,11 @@
+.overlay-custom-class {
```
Comment (JakeLaCombe) [inline packages/terra-abstract-modal/src/terra-dev-site/test/abstract-modal/AbstractModalTestStyles.scss]: Is there anything preventing us from doing CSS modules here? It doesn't add a lot of complexity, and it ensures we have styles that will not collide.

File: packages/terra-action-header/src/terra-dev-site/doc/example/BackActionHeader.jsx [hunk 8]
```diff
@@ -0,0 +1,18 @@
+/* eslint-disable no-alert */
+import React from 'react';
+// eslint-disable-next-line import/no-extraneous-dependencies, import/no-unresolved, import/extensions
```
Comment (JakeLaCombe) [inline packages/terra-action-header/src/terra-dev-site/doc/example/BackActionHeader.jsx]: Is this lint ignore still applicable?

File: packages/terra-paginator/package.json [hunk 9]
```diff
@@ -32,10 +32,14 @@
   "dependencies": {
     "classnames": "^2.2.5",
     "prop-types": "^15.5.8",
+    "react-router": "^4.2.0",
```
Comment (bjankord) [inline packages/terra-paginator/package.json]: Adding react-router as a dependency here feels weird.

Comment (mjhenkes) [reply packages/terra-paginator/package.json]: Agreed, but the examples would need to be re-worked to remove the dependency.

File: packages/terra-action-header/src/terra-dev-site/doc/example/BackActionHeader.jsx (hunk 8 above)
Comment (mjhenkes) [reply packages/terra-action-header/src/terra-dev-site/doc/example/BackActionHeader.jsx]: Yes, the next line is resolved by webpack and will fail the linter because it take into account webpack setup.

File: packages/terra-alert/src/terra-dev-site/doc/example/ActionAndDismissibleExample.jsx [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+      isDismissed: false,
+      actionButtonClickCount: 0,
+    };
+    this.handleDismiss = this.handleDismiss.bind(this);
+    this.actionFunc = this.actionFunc.bind(this);
+  }
+
+  handleDismiss() {
+    const newState = this.state;
+    newState.isDismissed = true;
```
Comment (JakeLaCombe) [inline packages/terra-alert/src/terra-dev-site/doc/example/ActionAndDismissibleExample.jsx]: This is actually changing the state I believe. If you are looking for this approach, you need to duplicate the object on line 18.

Comment (JakeLaCombe) [reply packages/terra-alert/src/terra-dev-site/doc/example/ActionAndDismissibleExample.jsx]: Actually, why not just do this?

```jsx
this.setState({
   isDismissed: true
});
```

Comment (mjhenkes) [reply packages/terra-alert/src/terra-dev-site/doc/example/ActionAndDismissibleExample.jsx]: This has been carried forward from the previous example. If we want to change it I think it would be best if we do that in a separate pull request.

File: packages/terra-paginator/package.json (hunk 9 above)
Comment (mjhenkes) [reply packages/terra-paginator/package.json]: Good call, I'm not sure where that came from then.

File: packages/terra-alert/src/terra-dev-site/doc/example/ActionAndDismissibleExample.jsx (hunk 10 above)
Comment (JakeLaCombe) [reply packages/terra-alert/src/terra-dev-site/doc/example/ActionAndDismissibleExample.jsx]: I noticed a lot of my comments were curtailed around old examples. I'm ok with leaving those off for future enhancements.

**Comment by JakeLaCombe** [comment]
I see that in the settings dropdown that I can still select the mock theme, even though the configuration doesn't use it anymore. Is there a trick I need to do in order to remove it?

**Comment by mjhenkes** [comment]
@JakeLaCombe, that was a mistake that emily pointed out. It's been updated, if you refresh your page.
screen shot 2018-05-29 at 10 18 42 am

**Comment by JakeLaCombe** [comment]
Sounds good. I ran it locally and saw this, but I may not have fetched recently, so it still showed up for me.

File: dev-site-config/site.config.js (hunk 3 above)
Comment (mjhenkes) [reply dev-site-config/site.config.js]: this config is used by node and no-longer transpiled, it's output is transpiled and use by the browser.

**Comment by saedar** [comment]
The `input-field` example isn't showing the button for the code pane.

**Comment by saedar** [comment]
There seems to be a lot wrong with the `signature` example.

* The canvas seems to extend off the right side.
* The sign-with-line dropdown should have the line width options but is currently a non-searchable search dropdown with no options.
* The sign-with-line dropdown is being overlapped by the code pane section.

File: packages/terra-signature/src/terra-dev-site/doc/example/SignatureExample.jsx [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+    this.setState(newState);
+  }
+
+  render() {
+    return (
+      <div style={{ height: '100px', width: '100%', border: '1px solid black' }} >
+        <Signature id="foo" lineWidth={this.state.lineWidth} lineSegments={this.state.lineSegments} />
+        <button onClick={this.handleClear}>Clear </button>
+        <button onClick={this.handleSingleLine}>Sign w/Line </button>
+        <Select
```
Comment (bjankord) [inline packages/terra-signature/src/terra-dev-site/doc/example/SignatureExample.jsx]: Need to update this to use the new select component/API

Comment (bjankord) [reply packages/terra-signature/src/terra-dev-site/doc/example/SignatureExample.jsx]: Resolved in this commit: https://github.com/cerner/terra-core/pull/1542/commits/e991c4c771f6ece77c547539d3bcd7dcb7cbd400

File: packages/terra-signature/src/terra-dev-site/doc/example/SignatureExample.jsx [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+  }
+
+  handleLineWidth(e) {
+    const newState = Object.assign({}, this.state, { lineSegments: [], lineWidth: parseInt(e.target.value, 10) });
+    this.setState(newState);
+  }
+
+  render() {
+    return (
+      <div style={{ height: '100px', width: '100%', border: '1px solid black' }} >
```
[the remark this answers is not in the document]
Comment (bjankord) [reply packages/terra-signature/src/terra-dev-site/doc/example/SignatureExample.jsx]: Resolved in this commit: https://github.com/cerner/terra-core/pull/1542/commits/e991c4c771f6ece77c547539d3bcd7dcb7cbd400

**Comment by mmalaker** [comment]
Functional spot check completed. The Terra Core site displays and functions as expected.
